src/mapping/query/function/split.rs

```diff
@@ -0,0 +1,247 @@
+use super::prelude::*;
+use bytes::Bytes;
+use lazy_static::lazy_static;
+use std::collections::BTreeMap;
+use std::ops::Deref;
+
+#[derive(Debug)]
+pub(in crate::mapping) struct SplitFn {
+    path: Box<dyn Function>,
+    pattern: Box<dyn Function>,
+    limit: Option<Box<dyn Function>>,
+}
+
+impl SplitFn {
+    #[cfg(test)]
+    pub(in crate::mapping) fn new(
+        path: Box<dyn Function>,
+        pattern: Box<dyn Function>,
+        limit: Option<Box<dyn Function>>,
+    ) -> Self {
+        Self {
+            path,
+            pattern,
+            limit,
+        }
+    }
+}
+
+lazy_static! {
+    static ref I_FLAG: Value = Value::from(Bytes::from("i"));
+    static ref G_FLAG: Value = Value::from(Bytes::from("g"));
+    static ref M_FLAG: Value = Value::from(Bytes::from("m"));
+}
+
+/// Create a regex from a map containing fields :
+/// pattern - The regex pattern
+/// flags   - flags including i => Case insensitive, g => Global, m => Multiline.
+fn regex_from_map(mut map: BTreeMap<String, Value>) -> Result<(regex::Regex, bool)> {
+    let pattern = map
+        .remove("pattern")
+        .ok_or_else(|| "Field is not a regular expression".to_string())?;
+
+    let flags = match map.remove("flags") {
+        None => Value::from(Vec::<Value>::new()),
+        Some(flags) => flags,
+    };
+
+    match (flags, pattern) {
+        (Value::Array(ref flags), Value::Bytes(ref bytes)) => {
+            let (global, insensitive, multi_line) =
+                flags
+                    .iter()
+                    .fold((false, false, false), |(g, i, m), flag| match flag {
+                        v if v == G_FLAG.deref() => (true, i, m),
+                        v if v == I_FLAG.deref() => (g, true, m),
+                        v if v == M_FLAG.deref() => (g, i, true),
+                        _ => (g, i, m),
+                    });
+
+            let pattern = String::from_utf8_lossy(&bytes);
+            let regex = regex::RegexBuilder::new(&pattern)
+                .case_insensitive(insensitive)
+                .multi_line(multi_line)
+                .build()
+                .map_err(|err| format!("invalid regex {}", err))?;
+            Ok((regex, global))
```

Comment:
If we use a newtype as suggested, we can have an `is_global(&self)` method, which removes the need to return a tuple in whatever function we create to return the compiled regex.